### ISSUE TITLE
optimise (search-index) memory usage

### ISSUE THREAD

**Issue by blahah** [blahah/yunodb#4]
See https://github.com/fergiemcdowall/search-index/issues/261 and https://github.com/fergiemcdowall/search-index-adder/issues/2

Some of the issues are addressed by the NLP pipeline which greatly reduces redundancy in the index. However, memory usage is still high.

A test case using [10,000 publication metadata records](https://gist.github.com/blahah/69621f1580aca56a94475252f2bcb273):

```
npm install --global yuno
wget https://git.io/vw48e
wget https://git.io/vw48L
yuno create --opts eupmc_db_opts.json memory_test eupmc_lite_10k.json
```

**Comment by eklem** [comment]
The new search-index streaming API does wonders here. Where I maxed out with 16 GB of memory before, I now manage with 1 GB of memory.

**Comment by blahah** [comment]
yes! see #6 - this is why we sponsored the streaming interface in search-index :)
